### PR TITLE
configure application with behats and symfony

### PR DESCRIPTION
**Pull request by AdamKasp**
Added behat steps with configuration and basic Symfony dependency injection configs

### CODE REVIEW

**Review by Rafikooo — APPROVED** [review]

File: features/using_predefined_date.feature [hunk 1]
```diff
@@ -0,0 +1,5 @@
+Feature: Using a predefined date
```
Comment (Zales0123) [inline features/using_predefined_date.feature:1]: There is no description of the feature :( Maybe something like

```
In order to be able to easily describe time-sensitive features
As a Developer
I would like to be able to use the predefined date
```

File: features/using_predefined_date.feature [hunk 2]
```diff
@@ -0,0 +1,5 @@
...
+
+    Scenario: Checking if Calendar provide date declared in setup
+        Given it is "2020-01-20" now
```
Comment (Zales0123) [inline features/using_predefined_date.feature]: When

File: features/using_predefined_date.feature [hunk 3]
```diff
@@ -0,0 +1,5 @@
...
+        Then Calendar provide the "2020-01-20" date
```
Comment (Zales0123) [inline features/using_predefined_date.feature]: Maybe `Then I should be able to use "2020-01-20" as my current date`?

File: features/using_predefined_date.feature [hunk 4]
```diff
@@ -0,0 +1,5 @@
+Feature: Using a predefined date
+
+    Scenario: Checking if Calendar provide date declared in setup
```
Comment (Zales0123) [inline features/using_predefined_date.feature]: `Using predefined date in time-sensitive scenario`?

File: src/Kernel.php [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+ */
+
+declare(strict_types=1);
+
+namespace Sylius\Calendar;
+
+use Symfony\Bundle\FrameworkBundle\Kernel\MicroKernelTrait;
+use Symfony\Component\HttpKernel\Kernel as BaseKernel;
+
+class Kernel extends BaseKernel
```
Comment (Zales0123) [inline src/Kernel.php]: Could be final :dancer:

File: tests/Behat/Context/Setup/CalendarContext.php [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+use Behat\Behat\Context\Context;
+
+final class CalendarContext implements Context
+{
+    private string $projectDirectory;
+
+    public function __construct(string $projectDirectory)
+    {
+        $this->projectDirectory = $projectDirectory;
+    }
```
Comment (Zales0123) [inline tests/Behat/Context/Setup/CalendarContext.php]: `public function __construct(private string $projectDirectory)` 🚀

File: tests/Behat/Provider/Calendar.php [hunk 7]
```diff
@@ -0,0 +1,44 @@
...
+
+use Sylius\Calendar\Provider\DateTimeProviderInterface;
+
+final class Calendar implements DateTimeProviderInterface
+{
+    private string $projectDirectory;
+
+    public function __construct(string $projectDirectory)
+    {
+        $this->projectDirectory = $projectDirectory;
```
Comment (Zales0123) [inline tests/Behat/Provider/Calendar.php]: Maybe we can set up the full path already here?

```php
$this->temporaryDatePath = $projectDirectory . '/temporartyDate.txt'
```

File: tests/Behat/Provider/Calendar.php [hunk 8]
```diff
@@ -0,0 +1,44 @@
...
+    }
+
+    public function now(): \DateTimeInterface
+    {
+        return $this->provideFakeDateIfSet();
+    }
+
+    private function provideFakeDateIfSet(): \DateTimeInterface
```
Comment (Zales0123) [inline tests/Behat/Provider/Calendar.php]: There is no need for this private function

File: tests/Behat/Provider/Calendar.php [hunk 9]
```diff
@@ -0,0 +1,44 @@
...
+    {
+        if (file_exists($this->projectDirectory . '/var/temporaryDate.txt')) {
+            $file = fopen($this->projectDirectory . '/var/temporaryDate.txt', 'r');
+            $dateTime = fgets($file);
+            fclose($file);
```
Comment (Zales0123) [inline tests/Behat/Provider/Calendar.php]: `file_get_contents` should do the trick

File: tests/Behat/Provider/Calendar.php [hunk 10]
```diff
@@ -0,0 +1,44 @@
...
+
+            return new \DateTimeImmutable($dateTime);
+        }
+
+        return new \DateTimeImmutable();
```
Comment (Zales0123) [inline tests/Behat/Provider/Calendar.php]: I believe if the file does not exist it should throw an exception - reading the date from the file is the only reason of this service, this way i pretends it works when it does not 🔥

File: tests/Behat/Provider/Calendar.php [hunk 11]
```diff
@@ -0,0 +1,44 @@
...
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Sylius\Calendar\Tests\Behat\Provider;
+
+use Sylius\Calendar\Provider\DateTimeProviderInterface;
+
+final class Calendar implements DateTimeProviderInterface
```
Comment (Zales0123) [inline tests/Behat/Provider/Calendar.php]: `FakeCalendar`? Because it is what it is 😄

**Review by Zales0123 — CHANGES_REQUESTED** [review]

**Review by Rafikooo — APPROVED** [review]

File: features/using_predefined_date.feature (hunk 3 above)
Comment (Zales0123) [reply features/using_predefined_date.feature]: Resolved - not fixed 💃

File: composer.json [hunk 12]
```diff
@@ -1,7 +1,7 @@
 {
   "name": "sylius/calendar",
   "type": "library",
-  "version": "v1.0-dev",
+  "version": "v0.0.1-dev",
```
Comment (GSadee) [inline composer.json:4]: I wonder if it shouldn't be `"v1.0-dev"` or at least `"v0.1-dev"`

**Review by GSadee — APPROVED** [review]

File: composer.json (hunk 12 above)
Comment (Zales0123) [reply composer.json:4]: `v0.1-dev` indeed 👍

**Review by Zales0123 — APPROVED** [review]

**Comment by Zales0123** [comment]
Thank you, @AdamKasp and @TheMilek! 🎉

**Pull request merged**